src/nvim/os/env.c

```diff
@@ -427,6 +427,31 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+  // XDG Base Directory expansion.
+  if (strncmp(name, "XDG_", 4) == 0) {
```

Comment:
Sneaking this into `vim_getenv` and making it do something totally different than what the caller asked, is unsettling.